# Data Cleaning and Preprocessing of Sentiment Dataset

## Objective
The aim of this project is to clean and preprocess a sentiment dataset by identifying missing values, removing duplicates, and standardizing inconsistent data formats using Python and Pandas.

In [2]:
#Import Libraries
import pandas as pd

In [3]:
#Load Dataset
df = pd.read_csv("Sentiment dataset.csv")
df.head()

,Unnamed: 0.1,Unnamed: 0,Text,Sentiment,Timestamp,User,Platform,Hashtags,Retweets,Likes,Country,Year,Month,Day,Hour
0,0,0,Enjoying a beautiful day at the park! ...,Positive,2023-01-15 12:30:00,User123,Twitter,#Nature #Park,15.0,30.0,USA,2023,1,15,12
1,1,1,Traffic was terrible this morning. ...,Negative,2023-01-15 08:45:00,CommuterX,Twitter,#Traffic #Morning,5.0,10.0,Canada,2023,1,15,8
2,2,2,Just finished an amazing workout! 💪 ...,Positive,2023-01-15 15:45:00,FitnessFan,Instagram,#Fitness #Workout,20.0,40.0,USA,2023,1,15,15
3,3,3,Excited about the upcoming weekend getaway! ...,Positive,2023-01-15 18:20:00,AdventureX,Facebook,#Travel #Adventure,8.0,15.0,UK,2023,1,15,18
4,4,4,Trying out a new recipe for dinner tonight. ...,Neutral,2023-01-15 19:55:00,ChefCook,Instagram,#Cooking #Food,12.0,25.0,Australia,2023,1,15,19


## Load Dataset
The dataset is loaded into a Pandas DataFrame for inspection and preprocessing.

## Dataset Overview
Inspecting the dataset structure, data types, and non-null counts.

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 732 entries, 0 to 731
Data columns (total 15 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Unnamed: 0.1  732 non-null    int64  
 1   Unnamed: 0    732 non-null    int64  
 2   Text          732 non-null    object 
 3   Sentiment     732 non-null    object 
 4   Timestamp     732 non-null    object 
 5   User          732 non-null    object 
 6   Platform      732 non-null    object 
 7   Hashtags      732 non-null    object 
 8   Retweets      732 non-null    float64
 9   Likes         732 non-null    float64
 10  Country       732 non-null    object 
 11  Year          732 non-null    int64  
 12  Month         732 non-null    int64  
 13  Day           732 non-null    int64  
 14  Hour          732 non-null    int64  
dtypes: float64(2), int64(6), object(7)
memory usage: 85.9+ KB


### Observation
The dataset contains 732 records and 15 columns. No missing values are immediately apparent from the non-null counts. The Timestamp column is stored as an object type and will require conversion to datetime format.

## Missing Value Analysis

In [5]:
#Missing Values Check
df.isnull().sum()

Unnamed: 0.1    0
Unnamed: 0      0
Text            0
Sentiment       0
Timestamp       0
User            0
Platform        0
Hashtags        0
Retweets        0
Likes           0
Country         0
Year            0
Month           0
Day             0
Hour            0
dtype: int64

### Observation
No missing values were found in any column of the dataset.

In [6]:
#Duplicate Check
df.duplicated().sum()

np.int64(0)

### Observation
No duplicate records were identified in the dataset.

In [7]:
#Remove Unnecessary Columns
df.drop(['Unnamed: 0', 'Unnamed: 0.1'], axis=1, inplace=True)

## Remove Unnecessary Columns

The dataset contains two unnecessary index columns (`Unnamed: 0` and `Unnamed: 0.1`) that were created during a previous CSV export. These columns do not contribute to the analysis and are removed to improve data quality.

In [8]:
#Standardize Text Fields
cols = ['User', 'Platform', 'Country', 'Sentiment']

for col in cols:
    df[col] = df[col].str.strip()

## Standardize Text Fields

Text-based columns are standardized by removing leading and trailing spaces. This ensures consistency in categorical values and prevents issues during analysis.

In [9]:
#Convert Timestamp
df['Timestamp'] = pd.to_datetime(df['Timestamp'])

## Convert Timestamp to Datetime Format

The `Timestamp` column is currently stored as a text (object) data type. It is converted to datetime format to enable proper date and time analysis.

In [10]:
#Save Cleaned Dataset
df.to_csv('cleaned_sentiment_dataset.csv', index=False)

## Save the Cleaned Dataset

After completing all preprocessing steps, the cleaned dataset is exported as a new CSV file for future analysis and visualization tasks.

In [11]:
#Final Preview
df.head()

,Text,Sentiment,Timestamp,User,Platform,Hashtags,Retweets,Likes,Country,Year,Month,Day,Hour
0,Enjoying a beautiful day at the park! ...,Positive,2023-01-15 12:30:00,User123,Twitter,#Nature #Park,15.0,30.0,USA,2023,1,15,12
1,Traffic was terrible this morning. ...,Negative,2023-01-15 08:45:00,CommuterX,Twitter,#Traffic #Morning,5.0,10.0,Canada,2023,1,15,8
2,Just finished an amazing workout! 💪 ...,Positive,2023-01-15 15:45:00,FitnessFan,Instagram,#Fitness #Workout,20.0,40.0,USA,2023,1,15,15
3,Excited about the upcoming weekend getaway! ...,Positive,2023-01-15 18:20:00,AdventureX,Facebook,#Travel #Adventure,8.0,15.0,UK,2023,1,15,18
4,Trying out a new recipe for dinner tonight. ...,Neutral,2023-01-15 19:55:00,ChefCook,Instagram,#Cooking #Food,12.0,25.0,Australia,2023,1,15,19


## Preview of the Cleaned Dataset

The first few rows of the cleaned dataset are displayed to verify that the preprocessing steps were applied successfully.

# Conclusion

The sentiment dataset was successfully cleaned and preprocessed using Python and Pandas. Data quality checks confirmed that there were no missing values or duplicate records. Unnecessary index columns were removed, text fields were standardized, and the Timestamp column was converted to datetime format. The cleaned dataset is now ready for further analysis and visualization.

In [4]:
df.to_csv("cleaned_sentiment_dataset.csv", index=False, encoding="utf-8")

In [5]:
import os

print(os.path.exists("cleaned_sentiment_dataset.csv"))
print(os.path.getsize("cleaned_sentiment_dataset.csv"))

True
171521


In [6]:
import os

print(os.path.exists("cleaned_sentiment_dataset.csv"))
print(os.path.getsize("cleaned_sentiment_dataset.csv"))

True
171521


In [7]:
import os

print(os.getcwd())

C:\Users\HP


In [8]:
print(os.path.exists("cleaned_sentiment_dataset.csv"))

True


In [9]:
print(os.path.getsize("cleaned_sentiment_dataset.csv"))

171521


In [10]:
pd.read_csv("cleaned_sentiment_dataset.csv").head()

,Unnamed: 0.1,Unnamed: 0,Text,Sentiment,Timestamp,User,Platform,Hashtags,Retweets,Likes,Country,Year,Month,Day,Hour
0,0,0,Enjoying a beautiful day at the park! ...,Positive,2023-01-15 12:30:00,User123,Twitter,#Nature #Park,15.0,30.0,USA,2023,1,15,12
1,1,1,Traffic was terrible this morning. ...,Negative,2023-01-15 08:45:00,CommuterX,Twitter,#Traffic #Morning,5.0,10.0,Canada,2023,1,15,8
2,2,2,Just finished an amazing workout! 💪 ...,Positive,2023-01-15 15:45:00,FitnessFan,Instagram,#Fitness #Workout,20.0,40.0,USA,2023,1,15,15
3,3,3,Excited about the upcoming weekend getaway! ...,Positive,2023-01-15 18:20:00,AdventureX,Facebook,#Travel #Adventure,8.0,15.0,UK,2023,1,15,18
4,4,4,Trying out a new recipe for dinner tonight. ...,Neutral,2023-01-15 19:55:00,ChefCook,Instagram,#Cooking #Food,12.0,25.0,Australia,2023,1,15,19
